### DISCUSSION TITLE
Docker e Broadlink Manager

### DISCUSSION THREAD

**Discussion by millenniumy2k** [indomus/forum#1223] [HUB: Home Assistant]
Ciao,
sto cercando di creare un pulsante per aprire il cancello (cosa che già faccio con il mio Broadlink) tramite HA.
Ho integrato Broadlink.
Il Broadlink Manager per windows non mi funziona, ho provato quindi a usare docker.
Tutto bene, ma il container creato utilizza un ip 192.168.65.3, mentre io sono nella classe 192.168.1.xx
Posso cambiare l'ip del container?
Grazie

**Comment by KC-inDomus** [comment]
scusa ma che c'entra, punta sull'IP della macchina ospitante docker sulla porta 7020, es http://192.168.1.x:7020

**Reply by millenniumy2k** [reply]
è quello che pensavo io... 
- il PC ha in IP fisso 192.168.1.10
- il container con HA (adesso è su Windows per fare le mie prove..) infatti è raggiungibile da quell'ip con la sua porta
- il container con Broadlink invece :  Running on http://192.168.65.3:7020/ (Press CTRL+C to quit)

**Reply by snakuzzo** [reply]
Immagino che vadano chiariti alcuni concetti lato network di Docker.
Dovresti esporre la porta 7020.
Come hai creato il container?

**Reply by millenniumy2k** [reply]
Grazie per il supporto intanto...
Seguendo questa guida:
https://indomus.it/guide/catturare-codici-infrarossi-e-radiofrequenza-usando-broadlink-manager-via-web-gui/

e quindi con il comando:
docker run -d --name=broadlinkmanager -v /broadlinkmanager:/opt/broadlinkmanager/datays --net=host --restart=unless-stopped techblog/broadlinkmanager

**Reply by snakuzzo** [reply]
Ma hai tutto su Windows? Non ho capito.
Considera che net host su windows e su mac non funziona

**Reply by snakuzzo** [reply]
Considera pure che su windows non so se di default hai il suo firewall attivo

**Reply by millenniumy2k** [reply]
Si, sto facendo tutte le mie prove su Windows con docker, firewall disattivato.

**Reply by snakuzzo** [reply]
Sì ma io ti sto dicendo che net host non funziona su windows.
Leggi [QUI](https://docs.docker.com/network/host/)

![1](https://user-images.githubusercontent.com/7205371/147261063-56b8cd61-c9fb-4071-b881-dca0f2c7b75a.png)

**Reply by snakuzzo** [reply]
Premesso questa cosa dell'host e soprattutto che quello che dici riguardo all'ip `192.168.65.3` mi sembra una cosa assurda (almeno per come l'hai descritta), ammesso che tu voglia esporre la 7020 del container dovresti ricrearlo in questo modo...

```
docker run -d --name=broadlinkmanager -v /broadlinkmanager:/opt/broadlinkmanager/datays -p 7020:7020 --restart=unless-stopped techblog/broadlinkmanager
```

**Reply by snakuzzo** [reply]
Appena provato in locale sul mio PC Linux (ma così come ti ho descritto)...

Qui creo il container

```
$ docker run -d --name=broadlinkmanager -p 7020:7020 --restart=unless-stopped techblog/broadlinkmanager
Unable to find image 'techblog/broadlinkmanager:latest' locally
latest: Pulling from techblog/broadlinkmanager
284055322776: Pull complete 
9498295476ab: Pull complete 
1a3b075365ac: Pull complete 
26bde36c9929: Pull complete 
aaab879b1a6d: Pull complete 
e7a8fbcde9e3: Pull complete 
59b7c8ff1f3c: Pull complete 
Digest: sha256:0d35ef8bf87daff76240dd06805321143e585ff17512a88616877ee9d39ae6bb
Status: Downloaded newer image for techblog/broadlinkmanager:latest
7ad58169c17fdfeb936f7bf6a48ad5c102d55f277b3617370d6ad88c52345753
```

Questo l'IP assegnato al mio PC

```
$ ifconfig wlp3s0 | grep 192
        inet 192.168.1.12  netmask 255.255.255.0  broadcast 192.168.1.255
```

Questo il container che sta girando

```
$ docker ps -a
CONTAINER ID   IMAGE                       COMMAND                  CREATED              STATUS                   PORTS                                       NAMES
7ad58169c17f   techblog/broadlinkmanager   "/usr/bin/python3 /o…"   About a minute ago   Up About a minute        0.0.0.0:7020->7020/tcp, :::7020->7020/tcp   broadlinkmanager
````

Questa una prova con telnet sull'IP LAN

```
$ telnet 192.168.1.12 7020
Trying 192.168.1.12...
Connected to 192.168.1.12.
Escape character is '^]'.
```

Come vedi funziona tutto...

![1](https://user-images.githubusercontent.com/7205371/147262093-351e8cea-23b1-4eec-8ce4-570e87f10983.png)

**Reply by millenniumy2k** [reply]
Grazie a tutti per il supporto, finalmente sono riuscito a farlo andare!

Non mi trova nessun Broadlink presente nella rete, ma questo è un'altro discorso ;-)

**Reply by snakuzzo** [reply]
Perché dovresti usare network host...e su windows non puoi

https://docs.docker.com/network/#network-drivers

**Comment by millenniumy2k** [comment]
non trova nulla..

![Immagine 2021-12-24 171311](https://user-images.githubusercontent.com/8614031/147364041-59e0566f-080f-4e2e-aeed-10e3a69a3957.png)
